juju/juju
# Move more cross model secret facade logic to use the service service

The remaining logic in the various cross model facades which access secret info from state is ported to use the secret service. The service just has no op methods for now,

Driveby - rename a params SecretRevisionChange Revision attr to LatestRevision to make its use more clear.

Also, add a table to record a remote cross model secret latest revision. The consuming model needs to know the latest revision of a cross model secret. For local secrets, this can be inferred, but for cross model, we need to record it.

From secret remote consumers. we don't record the unit uuid (unit doesn't exist in the model). We record the unit name, so update the table.

## QA steps

Just unit tests for now
Double check - bootstrap a controller and add a secret to the controller charm

## Links

**Jira card:** JUJU-5875
